File: solution/ts/week01-code-analysis/src/movie.ts
// Anemic domain anti-pattern: the entity Movie has no behavior
export class Movie {
    // Inconsistent naming
    public movieID: string;
    public title: string;
    public director: string;
    public unitPrice: number;
    public totalCopies: number;
    public borrowedCopies: number;

    constructor(movieID: string, title: string, director: string, totalCopies: number, unitPrice: number) {
        this.movieID = movieID;
        this.title = title;
        this.director = director;
        this.totalCopies = totalCopies;
        this.unitPrice = unitPrice;
        this.borrowedCopies = 0;
    }

    // Fragile implementation: a movie with a zero price cannot be sold?
    canSell(): boolean {
        // Incomplete rule: what if the unit price is negative?
        return this.unitPrice !== 0;
    }
}

File: solution/c#/week01-code-analysis/MovieStore/Movie.cs
namespace MovieStore
{
    //Anemic domain anti-pattern: the entity Movie has no behavior
    public class Movie(string movieId, string title, string director, int totalCopies, double unitPrice)
    {
        //Inconsistent naming
        public string MovieId { get; set; } = movieId;
        public string Title { get; set; } = title;
        public string Director { get; set; } = director;
        public double UnitPrice { get; set; } = unitPrice;
        public int TotalCopies { get; set; } = totalCopies;
        public int BorrowedCopies { get; set; } = 0;

        //Fragile implementation: a movie with a zero price cannot be sold?
        public bool CanSell()
        {
            //Incomplete rule: what is the unit price is negative?
            return UnitPrice != 0d;
        }
    }
}